state/unit.go

```diff
@@ -1267,11 +1266,9 @@ func (u *Unit) serviceAddress(scope string) (network.Address, error) {
 	if len(addresses) == 0 {
 		return network.Address{}, network.NoAddressError(scope)
 	}
-
 	getStrictPublicAddr := func(addresses []network.Address) (network.Address, bool) {
 		addr, ok := network.SelectPublicAddress(addresses)
-		ok = ok && addr.Scope == network.ScopePublic
-		return addr, ok
+		return addr, ok && (addr.Scope == network.ScopePublic || addr.Scope == network.ScopeCloudLocal)
```

Comment:
getStrictPublicAddr is not the correct name anymore

Reply:
After the latest discussion, it needs to stay as StrictPublic.
Since we will fall back to Private() which will return the CLoudLocal